ext/exif/exif.c

```diff
@@ -4279,11 +4291,128 @@ static bool exif_process_IFD_in_TIFF(image_info_type *ImageInfo, size_t dir_offs
 	return result;
 }
 
+static int exif_isobmff_parse_box(unsigned char *buf, isobmff_box_type *box)
+{
+	box->size = php_ifd_get32u(buf, 1);
+	buf += 4;
+	box->type = php_ifd_get32u(buf, 1);
+	if (box->size != 1) {
+		return 8;
+	}
+	buf += 4;
+	box->size = php_ifd_get64u(buf, 1);
+	return 16;
+}
+
+static void exif_isobmff_parse_meta(unsigned char *data, unsigned char *end, isobmff_item_pos_type *pos)
+{
+	isobmff_box_type box, item;
+	unsigned char *box_offset, *p, *p2;
+	int header_size, exif_id = -1, version, item_count, i;
+
+	for (box_offset = data + 4; box_offset < end; box_offset += box.size) {
+		header_size = exif_isobmff_parse_box(box_offset, &box);
+		if (box.type == FOURCC("iinf")) {
+			p = box_offset + header_size;
```

Comment:
is there any check that `p` doesn't get passed till `end`? I can see that `box_offset` is checked in the loop but somehow cannot see that `box_offset + header_size` cannot get past end.